### PR TITLE
cylc.job_host: more diagnostics on init failure

### LINKED ISSUE #1669: Better feedback on remote job submission problems needed.
The following problems are likely to afflict new users:
 * ~~failed to set up passwordless ssh to remote host~~
 * login scripts echo to stdout even in non-interactive shells (this is common, and arguably should not be considered an error)
 * (~~lack of access to cylc on the remote end~~; or access to the wrong version; other...?)?

all generate the following cryptic error, with no diagnostic information available:

```
2015-11-12T04:43:01Z ERROR - [foo.1] -2015-11-12T04:43:01Z [job-submit cmd] (prepare job file)
2015-11-12T04:43:01Z [job-submit ret_code] 1
2015-11-12T04:43:01Z [job-submit err] <host>: initialisation did not complete
```

### PR DESCRIPTION
**Pull request by matthewrmshin**
~~Address~~ Close #1669. On failure to initialise the suite run directory of remote a job host, it should now print out the failed command, its return code, and its standard output/error.

### CODE REVIEW

**Comment by matthewrmshin** [comment]
@hjoliver @arjclark please review.

**Comment by hjoliver** [comment]
Interestingly, your extra diagnostic info here reveals the source of the [#1669 bullet point 2](https://github.com/cylc/cylc/issues/1669#issuecomment-156233643) problem.  With an `echo` to stdout in `.tcshrc` (and same for `.bashrc` for bash login shell):
```
$ cylc run test --no-detach
2015-11-12T22:08:56Z ERROR - [foo.1] -2015-11-12T22:08:56Z [job-submit cmd] (prepare job file)
2015-11-12T22:08:56Z [job-submit ret_code] 1
2015-11-12T22:08:56Z [job-submit err]

wrh-2: initialisation did not complete:
COMMAND FAILED (1): scp -oBatchMode=yes -oConnectTimeout=10 -r /home/cylc_devel/cylc-run/test/cylc-suite-env 'wrh-2:$HOME/cylc-run/test/'
COMMAND STDERR: HELLO FROM .tcshrc

2015-11-12T22:08:56Z ERROR - [foo.1] -submission failed
```
i.e. this innocuous-looking `scp` command is failing!  It also fails when run at the command line.  Also, the startup script stdout appears in stderr, which is odd.

**Comment by hjoliver** [comment]
The reason, which was news to me, is that any output from login scripts for non-interactive shells breaks `scp` : http://www.openssh.com/faq.html#2.9

@matthewrmshin - this change is good in that it gives a much better clue to users about the cause of the problem.  Would you mind adding a short warning note to the user guide though, referencing the openssh FAQ item?

**Comment by hjoliver** [comment]
Review 1 - good; tests pass.

**Comment by matthewrmshin** [comment]
Added advice on impact of STDOUT on remote shell init to CUG.

File: tests/restart/bad-job-host/suite.rc [hunk 1]
```diff
@@ -29,6 +33,6 @@ rm -f "${CYLC_SUITE_WORK_DIR}/${CYLC_TASK_CYCLE_POINT}/t-remote/file"
             host = {{environ['CYLC_TEST_HOST']}}
     [[t-check-log]]
         script = """
-grep -q 'WARNING - garbage: initialisation did not complete' \
+timeout 30 grep -q 'ERROR - garbage: initialisation did not complete' \
```
Comment (arjclark) [inline tests/restart/bad-job-host/suite.rc]: What's the reasoning behind the addition of timeout here?

Comment (matthewrmshin) [inline tests/restart/bad-job-host/suite.rc]: It was pointless. Now removed.

**Comment by arjclark** [comment]
Looks good to me. No problems from test battery in my environment.

**Pull request merged**